pandas/_libs/tslibs/tzconversion.pyx

```diff
@@ -416,6 +416,9 @@ timedelta-like}
 
                 else:
                     delta_idx = bisect_right_i8(info.tdata, new_local, info.ntrans)
+                    if (shift_forward or shift_delta > 0) and \
+                       info.deltas[delta_idx-1] >= 0:
+                        delta_idx_offset = 1
```

Comment:
Not sure if it's safe to permanently change this to 1 for future iterations.

Might be safer to do something like

```
dst_offset = 0 if condition else 1
delta_idx = delta_idx - min(delta_idx_offset + dst_offset, 1)
```

Reply:
Right, make sense.

But there is this piece of code just before the loop to precompute `delta_idx_offset`. (Search for "Pre-compute delta_idx_offset"). I moved the whole block to here.

I don't know why it was precomputed. Was we assuming that there are just one time zone in the array?

Further more there is a recent PR around that block of code for a segfault problem.

Reply:
Oh misunderstood what you mean. Corrected.